modules/3d/include/opencv2/3d.hpp

```diff
@@ -2805,6 +2805,85 @@ CV_EXPORTS_W void loadMesh(const String &filename, OutputArray vertices, OutputA
 */
 CV_EXPORTS_W void saveMesh(const String &filename, InputArray vertices, InputArray normals, InputArrayOfArrays indices);
 
+//! Triangle fill settings
+enum class TriangleShadingType
+{
+    White  = 0, //!< a white color is used for the whole triangle
+    Flat   = 1, //!< a color of 1st vertex of each triangle is used
+    Shaded = 2  //!< a color is interpolated between 3 vertices with perspective correction
+};
+
+//! Face culling settings: what faces are drawn after face culling
+enum class TriangleCullingMode
+{
+    None = 0, //!< all faces are drawn, no culling is actually performed
+    CW   = 1, //!< triangles which vertices are given in clockwork order are drawn
+    CCW  = 2  //!< triangles which vertices are given in counterclockwork order are drawn
+};
+
+//! GL compatibility settings
+enum class TriangleGlCompatibleMode
+{
+    Disabled      = 0, //!< RGB and depth have their natural values and converted to internal formats if needed
+    InvertedDepth = 1  //!< RGB is natural, Depth is transformed from [-zNear; -zFar] to [0; 1]
+                       //!< by the following formula: \f$ \frac{z_{far} \left(z + z_{near}\right)}{z \left(z_{far} - z_{near}\right)} \f$ \n
+                       //!< In this mode the input/output depthBuf is considered to be in this format,
+                       //!< therefore it's faster since there're no conversions performed
+};
+
+/**
+ * @brief Structure to keep settings for rasterization
+ */
+struct CV_EXPORTS TriangleRasterizeSettings
+{
+    TriangleRasterizeSettings();
+
+    inline TriangleRasterizeSettings& setShadingType(TriangleShadingType st) { shadingType = st; return *this; }
+    inline TriangleRasterizeSettings& setCullingMode(TriangleCullingMode cm) { cullingMode = cm; return *this; }
+    inline TriangleRasterizeSettings& setGlCompatibleMode(TriangleGlCompatibleMode gm) { glCompatibleMode = gm; return *this; }
+
+    TriangleShadingType shadingType;
+    TriangleCullingMode cullingMode;
+    TriangleGlCompatibleMode glCompatibleMode;
+};
+
+
+/** @brief Renders a set of triangles on a depth and/or RGB image.
```

Comment:
BGR

Reply:
The renderer is colorspace-agnostic in fact, it just fills triangles with the values from `colors` array. I think this is the most convenient behaviour since it can be used to interpolate the data of arbitrary nature, e.g. coordinates or normals.
* I will change the vertex color format in test to BGR to get rid of extra `cvtColor` calls after the render
* For further compatibility purposes, I will add the formal limitation for colors to be in [0; 1]. The current code won't change its behavior though.